doc/Security.xml

```diff
@@ -947,6 +960,61 @@ Added certificate-based client authentication</revremark>
         </orderedlist>
       </section>
     </section>
+    <section>
+      <title>JWT-based client authorization</title>
+      <para>Unlike with HTTP Digest and Ws-USernameToken authentication, this specification defines how to associate clients with the different User Levels as defined in the ONVIF Core Specification. Devices are not expected to have the user credentials stored in their memory, instead they will verify that the user has been correctly authenticated by an external service based on Open ID Connect and authorize accessing different functions, based on the user level presented in  the supplied JWT embedded in the binary security token.</para>
+
+      <para>The inhalt of the binary security token is the base64-encoded JWT according to [RFC 7519].</para>
+      <para>The JOSE header shall declare that the encoded object is a JWT, and the JWT is a JWS that is MACed using the HMAC SHA-256. Additionally, it must include the kid claim as defined by [RFC 7515], which indicates the ID associated to the key used to sign the JWT. If the aboslute path of the JWK Set URL differns from  https://iss/.well-known/jwks.json, the header may include an optional jku claim.</para>
```

Comment:
I thought HS256 was mandatory because of https://www.rfc-editor.org/rfc/rfc7518#section-3 . Isn't it?

If it is not, we could use RS256/384/512.

Reply:
Yes implementations must support HS256 but other algos can be used. I prefer ES256 as the signature will be smaller than that of RS256.

Reply:
Since this would require supporting EC in the device, would it make sense to add support for EC to the keystore? I know they are two different things, but otherwise the security service would look schizophrenic.